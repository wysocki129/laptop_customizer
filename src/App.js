import React, { Component } from 'react';
import './App.css';
import ELFBranding from './ELFBranding/ELFBranding';
import Features from './Features/Features';
import Summary from './Summary/Summary';

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			selected: {
				Processor: {
					name:
						'17th Generation Intel Core HB (7 Core with donut spare)',
					cost: 700
				},
				'Operating System': {
					name: 'Ubuntu Linux 16.04',
					cost: 200
				},
				'Video Card': {
					name: 'Toyota Corolla 1.5v',
					cost: 1150.98
				},
				Display: {
					name:
						'15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
					cost: 1500
				}
			}
		};
	}

	updateFeature(feat, newItem) {
		const selected = Object.assign({}, this.state.selected);

		selected[feat] = newItem;

		this.setState({
			selected
		});
	}

	render() {
		return (
			<div className="App">
				<ELFBranding />
				<main>
					<Features
						features={this.props.features}
						selectedOptions={this.state.selected}
						handleUpdateFeature={(feat, newItem) =>
							this.updateFeature(feat, newItem)
						}
					/>
					<Summary selectedParts={this.state.selected} />
				</main>
			</div>
		);
	}
}

export default App;
