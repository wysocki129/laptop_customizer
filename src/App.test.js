import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

describe('App Component', () => {
	it.skip('renders without crashing the app', () => {
		shallow(<App />);
	});
});
