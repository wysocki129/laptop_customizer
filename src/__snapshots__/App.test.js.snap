// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component displays features propperly 1`] = `
<div
  className="App"
>
  <ELFBranding />
  <main>
    <Features
      features={
        Object {
          "Display": Array [
            Object {
              "cost": 1500,
              "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
            },
            Object {
              "cost": 1400,
              "name": "15.3\\" HGTV (3840 x 2160) Home makeover edition",
            },
          ],
          "Operating System": Array [
            Object {
              "cost": 200,
              "name": "Ubuntu Linux 16.04",
            },
            Object {
              "cost": 300,
              "name": "Bodhi Linux",
            },
          ],
          "Processor": Array [
            Object {
              "cost": 700,
              "name": "17th Generation Intel Core HB (7 Core with donut spare)",
            },
            Object {
              "cost": 1200,
              "name": "Professor X AMD Fire Breather with sidewinder technology",
            },
          ],
          "Video Card": Array [
            Object {
              "cost": 1150.98,
              "name": "Toyota Corolla 1.5v",
            },
            Object {
              "cost": 1345,
              "name": "Mind mild breeze 2000",
            },
          ],
        }
      }
      handleUpdateFeature={[Function]}
      selectedOptions={
        Object {
          "Display": Object {
            "cost": 1500,
            "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
          },
          "Operating System": Object {
            "cost": 200,
            "name": "Ubuntu Linux 16.04",
          },
          "Processor": Object {
            "cost": 700,
            "name": "17th Generation Intel Core HB (7 Core with donut spare)",
          },
          "Video Card": Object {
            "cost": 1150.98,
            "name": "Toyota Corolla 1.5v",
          },
        }
      }
    />
    <Summary
      selectedParts={
        Object {
          "Display": Object {
            "cost": 1500,
            "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
          },
          "Operating System": Object {
            "cost": 200,
            "name": "Ubuntu Linux 16.04",
          },
          "Processor": Object {
            "cost": 700,
            "name": "17th Generation Intel Core HB (7 Core with donut spare)",
          },
          "Video Card": Object {
            "cost": 1150.98,
            "name": "Toyota Corolla 1.5v",
          },
        }
      }
    />
  </main>
</div>
`;
